pkg/kubecost/summaryallocation.go

```diff
@@ -0,0 +1,1155 @@
+
+				sa.CPUCost += idle.CPUCost * cpuCoeff
+				sa.GPUCost += idle.GPUCost * gpuCoeff
+				sa.RAMCost += idle.RAMCost * ramCoeff
+			}
+		}
+	}
+
+	// (#) Apply idle filtration
+	if filteredTotals != nil {
+		for idleKey := range aggSet.idleKeys {
+			ia := aggSet.SummaryAllocations[idleKey]
+
+			var key string
+			if options.IdleByNode {
+				key = ia.Properties.Node
+			} else {
+				key = ia.Properties.Cluster
+			}
+
+			// Percentage of idle that should remain after filters are applied,
+			// which equals the proportion of filtered-to-actual cost.
+			cpuFilterCoeff := 0.0
+			if allocTotals[key].CPUCost > 0.0 {
+				cpuFilterCoeff = filteredTotals[key].CPUCost / allocTotals[key].CPUCost
+			}
+
+			gpuFilterCoeff := 0.0
+			if allocTotals[key].RAMCost > 0.0 {
+				gpuFilterCoeff = filteredTotals[key].RAMCost / allocTotals[key].RAMCost
+			}
+
+			ramFilterCoeff := 0.0
+
+			if allocTotals[key].RAMCost > 0.0 {
+				ramFilterCoeff = filteredTotals[key].RAMCost / allocTotals[key].RAMCost
+			}
+
+			ia.CPUCost *= cpuFilterCoeff
+			ia.GPUCost *= gpuFilterCoeff
+			ia.RAMCost *= ramFilterCoeff
+		}
+	}
+
+	// (#) Convert shared hourly cost into a cumulative allocation to share
+	for name, cost := range options.SharedHourlyCosts {
+		if cost > 0.0 {
+			hours := sas.Window.Hours()
+
+			// If set ends in the future, adjust hours accordingly
+			diff := time.Since(*sas.Window.End())
+			if diff < 0.0 {
+				hours += diff.Hours()
+			}
+
+			totalSharedCost := cost * hours
+
+			shareSet.Insert(&SummaryAllocation{
+				Name:       fmt.Sprintf("%s/%s", name, SharedSuffix),
+				Start:      *sas.Window.Start(),
+				End:        *sas.Window.End(),
+				SharedCost: totalSharedCost,
+			})
+		}
+	}
+
+	// (#) Distribute shared resources.
+	if len(shareSet.SummaryAllocations) > 0 {
+		// TODO deprecate ShareEven or accept new definition?
+		// TODO consider sharing by Cluster or Node?
+
+		sharingCoeffDenominator := 0.0
+		for _, rt := range allocTotals {
+			if options.ShareSplit == ShareEven {
+				sharingCoeffDenominator += float64(rt.Count)
+			} else {
+				sharingCoeffDenominator += rt.TotalCost()
+			}
+		}
+
+		// Do not include the shared costs, themselves, when determining
+		// sharing coefficients.
+		for _, rt := range sharedResourceTotals {
+			if options.ShareSplit == ShareEven {
+				sharingCoeffDenominator -= float64(rt.Count)
+			} else {
+				sharingCoeffDenominator -= rt.TotalCost()
+			}
+		}
+
+		// Do not include the unmounted costs when determining sharing
+		// coefficients becuase they do not receive shared costs.
+		if options.ShareSplit == ShareEven {
+			if totalUnmountedCost > 0.0 {
+				sharingCoeffDenominator -= 1.0
+			}
+		} else {
+			sharingCoeffDenominator -= totalUnmountedCost
+		}
+
+		// Compute sharing coeffs by dividing the thus-far accumulated
+		// numerators by the now-finalized denominator.
+		for key := range sharingCoeffs {
+			sharingCoeffs[key] /= sharingCoeffDenominator
```

Comment:
Would we encounter a problem if this division results in a NaN or other edge-case float?

Reply:
The assumption is that the case where `sharingCoeffDenominator` is `0.0` occurs only when the set it empty, and then you don't even get here. But that's unsafe, so I'll address it.